packages/ruleset/test/operation-id-naming-convention.test.js

```diff
@@ -12,24 +12,186 @@ describe('Spectral rule: operation-id-naming-convention', () => {
-      expect(r.severity).toBe(expectedSeverity);
-      expect(r.path.join('.')).toBe('paths./v1/drinks.post.operationId');
+      expect(results).toHaveLength(0);
+    });
+
+    it('path has multiple path params, get', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      const newGet = makeCopy(testDocument.paths['/v1/drinks/{drink_id}'].get);
+      newGet.operationId = 'get_glass';
+      testDocument.paths['/v1/drinks/{drink_id}/glasses/{glass_id}'] = {
+        get: newGet
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('path has single path param not at end, get', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      const newGet = makeCopy(testDocument.paths['/v1/drinks/{drink_id}'].get);
+      newGet.operationId = 'anything_goes';
+      testDocument.paths['/v1/drinks/{drink_id}/glass'] = {
+        get: newGet
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('path ends with path param, post', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks/{drink_id}'].post = {
+        operationId: 'create_drink'
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('no path param, put', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks'].put = {
+        operationId: 'replace_drinks'
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('path ends with path param, put', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks/{drink_id}'].put = {
+        operationId: 'replace_drink'
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('path has single path param not at end, put', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      // This should not fail as it is not considered to be "resource-oriented".
+      testDocument.paths['/v1/drinks/{drink_id}/glasses'] = {
+        put: {
+          operationId: 'anything_goes'
+        }
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('path has single path param not at end, put', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks/{drink_id}/glasses'] = {
+        put: {
+          operationId: 'set_drink_glass'
+        }
+      };
+
+      // Add an extra path to make the API resource-oriented.
+      testDocument.paths['/v1/drinks/{drink_id}/glass/{glass_id}'] = {
```

Comment:
I think the path segment here was intended to be `/glasses/` but it's a bit academic. The scenario for `set_` actually _shouldn't_ have a subpath like, so this test wouldn't really check against the `set_` scenario outlined in the Handbook.